plutarch-test/src/Plutarch/ScriptsSpec.hs

```diff
@@ -0,0 +1,225 @@
+module Plutarch.ScriptsSpec (
+  authorizedValidator,
+  authorizedPolicy,
+  authorizedStakeValidator,
+  authValidatorCompiled,
+  validatorEncoded,
+  validatorHashEncoded,
+  authValidatorHash,
+  authStakeValidatorCompiled,
+  stakeValidatorEncoded,
+  authStakeValidatorHash,
+  stakeValidatorHashEncoded,
+  authPolicyCompiled,
+  policyEncoded,
+  policySymEncoded,
+  authPolicySymbol,
+  spec,
+) where
+
+import System.FilePath ((</>))
+
+import Data.Text (Text)
+
+import Data.Aeson.Extras (encodeSerialise)
+import qualified Plutus.V1.Ledger.Api as Plutus
+import qualified Plutus.V1.Ledger.Crypto as Plutus
+
+import Plutarch (ClosedTerm, POpaque, popaque)
+import Plutarch.Api.V1 (
+  PScriptContext,
+  mintingPolicySymbol,
+  mkMintingPolicy,
+  mkStakeValidator,
+  mkValidator,
+  stakeValidatorHash,
+  validatorHash,
+  type PMintingPolicy,
+  type PStakeValidator,
+  type PValidator,
+ )
+import Plutarch.Api.V1.Crypto (PPubKey, PPubKeyHash, PSignature (PSignature))
+import Plutarch.Builtin (pasByteStr)
+import Plutarch.Prelude
+import Plutarch.Test (PlutarchGolden (PrintTerm), golden)
+import Test.Syd (Spec, describe, it, pureGoldenTextFile)
+
+spec :: Spec
+spec = do
+  describe "scripts" $ do
+    describe "auth_validator" $ do
+      golden PrintTerm authValidatorTerm
+      it "serialization" $
+        pureGoldenTextFile
+          ("goldens" </> "scripts.auth_validator.plutus.golden")
+          validatorEncoded
```

Comment:
You can use `getTestDescriptionPath` to obviate hardcoding the path here, which is error-prone. It would be even better if we just used `goldens` or `golden` from `Plutarch.Test` here.  Maybe expand `PlutarchGolden` to include `ScriptEncoding`?

Reply:
I went ahead and added `getGoldenFilePrefix` & `goldenFilePath` to make this a bit easier for tests.

Reply:
Don't think it's necessary to expand the existing golden tests with encoding, though